src/chainlink/VRFSubscription.ts

```diff
@@ -0,0 +1,106 @@
+import { writeContract } from "../core/helpers/util";
+
+
+export class VRFSubscription {
+    public id: string = "";
+    public balance: string = "";
+    public requestCount: string = "";
+    public owner: string = "";
+    public consumers: string[] = [];
```

Comment:
How is consumers array populated?

Reply:
There is a line in getSubscription() method of VRFCoordinator
`subscription.consumers = Array.from(callRequest.return_values[3]);`